File: P1_S2/src/p1_s2/Cliente.java

package p1_s2;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.Random;
import java.util.Set;

/**
 * 
 * @author Raquel Molina Reche (GH: rmr00),
 * Pablo Jiménez Jiménez (GH: pablojj1808)
 */
public class Cliente {
    private Evento evento;

    
    Cliente(AbstractFactory af) {
        //Se generan el numero de entradas del evento aleatoriamente
        int ne = generarNumeroEntradas();

        //Se crea el evento
        this.evento  = af.crearEvento(ne);
        
        //Se crean tantos usuarios como entradas tenga el evento
        for(int i=0; i<ne; i++) {
            evento.addPublico(af.crearUsuario(i+1));
        }
    }
    
    public static int generarNumeroEntradas(){
        Random rand = new Random();
        int entradas = rand.nextInt(20)+20;
        return  entradas;  
    }

    void hacerCosas() {      
        evento.start();
        
    }

}

File: P1_S2_rb/Cliente.rb

class Cliente

    attr_reader :usuario

    def initialize abs_factory
        @abs_factory = abs_factory

        @evento = @abs_factory.crear_evento 
        
        for i in 0..@evento.n_entradas
            @evento.add_publico(@abs_factory.crear_usuario)
        end
    end

    def hacer_cosas 
        @evento.empezar
    end

end